crates/common/download/src/download.rs

```diff
@@ -931,7 +927,7 @@ mod tests {
         let target_path = target_dir_path.path().join("test_download");
         let mut downloader = Downloader::new(target_path, None, CloudRootCerts::from([]));
         downloader.set_backoff(ExponentialBackoff {
-            current_interval: Duration::ZERO,
+            max_elapsed_time: Some(Duration::ZERO),
```

Comment:
Yes. As now the connect errors are considered transient, it causes DNS resolution failure to be considered transient, which caused the test to hang instead of immediately returning.
I think the intention behind setting `current_interval` to zero would be to disable retries in the test, but it wasn't actually doing anything, and only `max_elapsed_time` set to zero actually disables the retries.

But another question is should DNS resolution failure be considered transient. On one hand we can timeout during the query because the network is down, then we should retry, but if the network is up and the domain does not exist, then we should return ASAP.

I'll check if we can extract information from `reqwest::Error` so that we can distinguish between these 2 cases.

Reply:
> But another question is should DNS resolution failure be considered transient.

Yes DNS resolution errors are transient and should be retried, see https://github.com/thin-edge/thin-edge.io/issues/2321